### PR TITLE
Langfuse Wrapper

### PR DESCRIPTION
**Pull request by JUSTSUJAY**
1. Added Langfuse wrapper around the Openai Instance
2. Tested the tracing

### CODE REVIEW

File: configs/app.config.js [hunk 1]
```diff
@@ -7,5 +7,9 @@ module.exports = {
     },
     dbConfig: {
         PATH: "/tmp/database.db",
-    }
+    },
+    openaiAPIKey:process.env.OPENAI_API_KEY, 
+    langfusePublicKey:process.env.LANGFUSE_PUBLIC_KEY, 
+    langfuseSecretKey:process.env.LANGFUSE_SECRET_KEY,
+    langfuseBaseURL:process.env.LANGFUSE_BASE_URL
```
Comment (shashanksingh2002) [inline configs/app.config.js:14]: Please also rename the keys to be in caps lock

eg: API_KEY, SECRET_KEY

File: helpers/openaiInstance.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const { 
+    openaiAPIKey, 
+    langfusePublicKey, 
+    langfuseSecretKey,
+    langfuseBaseURL
+} = require('../configs/app.config')
+
+const openaiClient = new OpenAI({
+    apiKey: openaiAPIKey,
+})
```
Comment (shashanksingh2002) [inline helpers/openaiInstance.js:12]: Can we please instantiate this before the server starts? Please refer ASESSMENT-API to see how we have instantiated redis and node-cache

File: services/code.service.js [hunk 3]
```diff
@@ -7,8 +7,7 @@ const path = require('path')
 const sqlite3 = require('sqlite3').verbose()
 const { PYTHON, PROMPTV1, PROMPTV2, PROMPTV3 } = require('../enums/supportedLanguages')
 const logger = require('../loader').helpers.l
-const OpenAI = require('openai')
-const openai = new OpenAI()
+const { openai } = require('../helpers/openaiInstance')
```
Comment (shashanksingh2002) [inline services/code.service.js:10]: Has this flow been tested for subjective evaluations? As they use openAI calls to evaluate answers

**Review by shashanksingh2002 — CHANGES_REQUESTED** [review]
Please resolve the added comments

File: configs/app.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (JUSTSUJAY) [reply configs/app.config.js:14]: Do you mean to separate them with linespace and corresponding comments?

File: services/code.service.js (hunk 3 above)
Comment (JUSTSUJAY) [reply services/code.service.js:10]: Made the requested changes, and rebuilt the docker container, and the post requests were successfully sent via Postman and have been traced on Langfuse.

**Review by shashanksingh2002 — APPROVED** [review]
LGTM

**Pull request merged**